src/improvements/doc/NEW_TEMPLATE_GUIDE.md

```diff
@@ -0,0 +1,273 @@
+# New Template Creation Guide
+
+This guide explains how to create new Solidity templates for the superchain task system. Templates are the foundation for standardizing and securing task execution across different networks.
+
+## Quick Start
+
+To scaffold a new template:
+
+```bash
+cd src/improvements/
+just new template
+```
+
+This will create a new Solidity file in `src/improvements/template/` with the basic structure required.
+
+## Template Structure
+
+### 1. Required Components
+
+Every template must implement the following functions:
+
+```solidity
+// Required data structures
+struct TaskConfig {
+    // Template-specific configuration parameters
+}
+
+// Storage mapping
+mapping(uint256 => TaskConfig) public taskConfig;
+
+// Core functions
+function safeAddressString() internal pure override returns (string memory);
+function _taskStorageWrites() internal pure override returns (string[] memory);
+function _templateSetup(string memory taskConfigFilePath) internal override;
+function _build(uint256 chainId) internal override;
+function _validate(uint256 chainId) internal override;
+```
+
+The struct and mapping are optional and can be customized based on task requirements. As a general rule, task developers should think of everything as a template, hence the chainId flag to the build and validate functions.
+
+### 2. Function Implementations
+
+#### safeAddressString
+Returns the name of the multisig address from the superchain-registry:
+```solidity
+function safeAddressString() internal pure override returns (string memory) {
+    return "OptimismMultisig";
+}
+```
+
+#### _taskStorageWrites
+Lists addresses from superchain-registry whose storage will be modified:
+```solidity
+function _taskStorageWrites() internal pure override returns (string[] memory) {
+    return ["SystemConfig"];
```

Comment:
Again, lets use an actual example here. We don't ever write storage on implementation contracts. Therefore `SystemConfig` should be `SystemConfigProxy` here. 
```suggestion
    return ["SystemConfigProxy"];
```